app/src/lib/git/stash.ts

```diff
@@ -69,17 +69,41 @@ export async function getDesktopStashEntries(
   return stashEntries
 }
 
+export async function getLastStashEntry(
+  repository: Repository,
+  branchName: string
+) {
+  const entries = await getDesktopStashEntries(repository)
+
+  return entries.find(stash => stash.branchName === branchName) || null
```

Comment:
This function relies on the sort order of `git stash list`. Since that ordering is reverse-chronological order, `find` should be sufficient to return the **first** match stash entry.

Reply:
📝 this also seems worth a code comment so it's not easily forgotten once this PR is merged

Reply:
Added to the docs for `getDesktopStashEntries`.